pytorch/pytorch_checkpoint.py

```diff
@@ -84,12 +90,14 @@ def objective(trial):
     optimizer = getattr(optim, optimizer_name)(model.parameters(), lr=lr)
 
     trial_number = RetryFailedTrialCallback.retried_trial_number(trial)
-    trial_checkpoint_dir = os.path.join(CHECKPOINT_DIR, str(trial_number))
-    checkpoint_path = os.path.join(trial_checkpoint_dir, "model.pt")
-    checkpoint_exists = os.path.isfile(checkpoint_path)
 
-    if trial_number is not None and checkpoint_exists:
-        checkpoint = torch.load(checkpoint_path)
+    if trial_number is not None:
+        study = optuna.load_study(study_name="pytorch_checkpoint", storage="sqlite:///example.db")
+        artifact_id = study.trials[trial_number].user_attrs["artifact_id"]
```

Comment:
If the process is terminated before the first checkpoint, the artifact will not be saved, so check if it exists.